### ISSUE TITLE
dexie-export-import: Out-of-memory Crash On Import

### ISSUE THREAD

**Issue by travddm** [dfahlander/Dexie.js#872]
Dexie.js version 2.0.4
dexie-export-import version 1.0.0-beta.12
-> Using default options

While attempting to import a very large (~1.5 GB) previously exported database, chrome repeatedly ran out of memory and crashed. The very same code works fine for smaller files. It is running in a chrome worker, and given a Blob that is properly formatted:

```
const db = await Dexie.import(blob, {
    progressCallback: progressCallback
});
```

After creating an Allocation sampling profile, it became apparent that there was a significantly larger portion of memory being used in comparison to the file size. It rose to ~2GB before crashing.

Here are the most significant profiler results:
![image](https://user-images.githubusercontent.com/12009527/61313055-a03a0c00-a7c7-11e9-8b3d-b8e75b6a56b0.png)

I'll attempt to fix this and submit a PR.

**Comment by appletomato** [comment]
I can confirm this issue. For me, I get the out of memory crash for a database that is about 800 MB in size.

**Comment by travddm** [comment]
I've fixed the issue and submitted a PR (#874).

In short: the json streaming parser was not making objects available (and thus not allowing them to be released from memory) until they're completely finished parsing, which made it impossible to read partial objects without parsing the entire json. This was causing `loadUntilWeGotEnoughData()` to parse the entire json, where it really only needed a tiny amount of information from the beginning.

**Comment by dfahlander** [comment]
Fixed in PR #872, released in dexie-export-import@1.0.0-beta.13
